hazelcast/src/main/java/com/hazelcast/spi/impl/classicscheduler/PartitionOperationThread.java

```diff
@@ -0,0 +1,30 @@
+package com.hazelcast.spi.impl.classicscheduler;
+
+import com.hazelcast.instance.HazelcastThreadGroup;
+import com.hazelcast.instance.NodeExtension;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.spi.impl.OperationHandler;
+
+import java.util.Queue;
+import java.util.concurrent.BlockingQueue;
+
+/**
+ * An {@link OperationThread} that executes Operations for a particular partition, e.g. a map.get operation.
+ */
+public class PartitionOperationThread extends OperationThread {
+
+    private final OperationHandler[] partitionOperationHandlers;
+
+    public PartitionOperationThread(String name, int threadId, BlockingQueue workQueue,
+                                    Queue priorityWorkQueue, ILogger logger,
+                                    HazelcastThreadGroup threadGroup, NodeExtension nodeExtension,
+                                    OperationHandler[] partitionOperationHandlers) {
+        super(name, threadId, workQueue, priorityWorkQueue, logger, threadGroup, nodeExtension);
+        this.partitionOperationHandlers = partitionOperationHandlers;
+    }
+
+    @Override
+    public OperationHandler getOperationHandler(int partitionId) {
+        return partitionOperationHandlers[partitionId];
```

Comment:
My bad. it is good code.. Each partition will have its own operation-handler. This is needed for the progressive scheduler to allow for workstealing on a partitionlevel.